internal/pkg/agent/application/actions/handlers/handler_action_policy_change.go

```diff
@@ -317,9 +329,27 @@ func (h *PolicyChangeHandler) handlePolicyChange(ctx context.Context, c *config.
 		return fmt.Errorf("applying FleetClientConfig: %w", err)
 	}
 
+	if loggingHasChanged {
```

Comment:
This restart is only for the logging output correct? A log level change does not require this correct?

Reply:
You should clarify this in a comment to ensure future readers do not interpret this as broader than it needs to be.

Also you should explain exactly why it won't work if you don't restart, so that we know if/when we can remove this later if we for some reason need to.

Reply:
> This restart is only for the logging output correct? A log level change does not require this correct?

Yes.

I'll better document it.